### ISSUE TITLE
Update copy of gnulib (`gl/`) to fix musl build

### ISSUE THREAD

**Issue by thesamesam** [monitoring-plugins/monitoring-plugins#1831]
Hi folks,

Would it be possible to update the copy of gnulib in `gl/`? It should fix various issues fixed upstream in gnulib since the last sync, including building on musl (a lot of changes went into the gettext support for this), see e.g. https://bugs.gentoo.org/892645.

Thanks!

**Comment by RincewindsHat** [comment]
Hi @thesamesam,
That is a good idea and I tried it at least two times in the last two years, but it always broke. Maybe I give it a third try :-)
But, of course, if you know how to handle that, a PR would be welcome

**Comment by thesamesam** [comment]
Happy to give it a whack and then if I get stuck, we can try figure something out 🧪

**Comment by RincewindsHat** [comment]
hm, https://github.com/RincewindsHat/monitoring-plugins/actions/runs/4054745948/jobs/6976955390

**Comment by thesamesam** [comment]
Gets a bit further with:
```
diff --git a/plugins/Makefile.am b/plugins/Makefile.am
index 3fde54d6..555627db 100644
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -51,7 +51,7 @@ noinst_LIBRARIES = libnpcommon.a
 libnpcommon_a_SOURCES = utils.c netutils.c sslutils.c runcmd.c \
        popen.c utils.h netutils.h popen.h common.h runcmd.c runcmd.h

-BASEOBJS = libnpcommon.a ../lib/libmonitoringplug.a ../gl/libgnu.a
+BASEOBJS = libnpcommon.a ../lib/libmonitoringplug.a ../gl/libgnu.a $(LIB_CRYPTO)
 NETOBJS = $(BASEOBJS) $(EXTRA_NETOBLS)
 NETLIBS = $(NETOBJS) $(SOCKETLIBS)
 SSLOBJS = $(BASEOBJS) $(NETLIBS) $(SSLLIBS)
```

Then I get:
```
check_disk.c: In function ‘get_path_stats’:
check_disk.c:1059:26: error: ‘struct fs_usage’ has no member named ‘fsu_favail’; did you mean ‘fsu_bavail’?
 1059 |   p->inodes_free  = fsp->fsu_favail > fsp->fsu_ffree ? 0 : fsp->fsu_favail;
      |                          ^~~~~~~~~~
      |                          fsu_bavail
check_disk.c:1059:65: error: ‘struct fs_usage’ has no member named ‘fsu_favail’; did you mean ‘fsu_bavail’?
 1059 |   p->inodes_free  = fsp->fsu_favail > fsp->fsu_ffree ? 0 : fsp->fsu_favail;
      |                                                                 ^~~~~~~~~~
      |                                                                 fsu_bavail
make[3]: *** [Makefile:2584: check_disk.o] Error 1
make[3]: Leaving directory '/home/sam/git/monitoring-plugins/plugins'
```
... which looks like an API change that needs to be adapted to.

**Comment by RincewindsHat** [comment]
@thesamesam That helped, now it fails later ;-)

**Comment by RincewindsHat** [comment]
https://github.com/monitoring-plugins/monitoring-plugins/pull/1832 at least it compiles now

**Comment by thesamesam** [comment]
Built OK on glibc. Trying musl...

**Comment by waja** [comment]
https://bugs.gentoo.org/892645#c6 has also some information about it

**Comment by RincewindsHat** [comment]
Resolved with #1832
Thank you @thesamesam for the huge help here :-)

**Comment by RincewindsHat** [comment]
Should be fixed with #1832
